topics/bloomfilters.md

```diff
@@ -0,0 +1,108 @@
+---
+title: "Bloom Filters"
+description: >
+    Introduction to Bloom Filters
+---
+
+The bloom filter data type is taken from a [separate module](https://github.com/valkey-io/valkey-bloom) that users will need to install in order to use. 
+
+Bloom filters are a space efficient probabilistic data structure that allows checking whether an element is member of a set. False positives are possible, but it guarantees no false negatives.
+
+## Basic Bloom commands
+
+* `BF.ADD` adds an item to a bloom filter
+* `BF.CARD` returns the cardinality of a bloom filter
+* `BF.EXISTS` checks if an item has been added to a bloom filter
+* `BF.INFO` returns information about a bloom filter
+
+See the [complete list of bloom filter commands](../commands/#bloom).
+
+## Common use cases for bloom filters
+
+### Financial fraud detection
+
+Bloom filters can help answer the question "Has this card been flagged as stolen?", use a bloom filter that has cards reported stolen added to it. Check a card on use that it is not present in the bloom filter. If it isn't then the card is not marked as stolen, if present then a check to the main database can happen or deny the purchase.
+
+### Ad placement
+
+Bloom filters can help answer the following questions to advertisers:
+* Has the user already seen this ad?
+* Has the user already bought this product?
+
+Use a Bloom filter for every user, storing all bought products. The recommendation engine can then suggest a new product and checks if the product is in the user's Bloom filter.
+
+* If no, the ad is shown to the user and is added to the Bloom filter.
+* If yes, the process restarts and repeats until it finds a product that is not present in the filter.
+
+### Check if URL's are malicious
+
+Bloom filters can answer the question "is a URL malicious?". Any URL inputted would be checked against a malicious URL bloom filter. 
+
+* If no then we allow access to the site
+* If yes then we can deny access or perform a full check of the URL
+
+### Check if a username is taken
+
+Bloom filters can answer the question: Has this username/email/domain name/slug already been used?
+
+For example for usernames. Use a Bloom filter for every username that has signed up. A new user types in the desired username. The app checks if the username exists in the Bloom filter.
+
+* If no, the user is created and the username is added to the Bloom filter.
+* If yes, the app can decide to either check the main database or reject the username.
+
+## Default bloom properties
+
+Capacity - 100
+
+Error rate - 0.01
+
+Expansion - 2
```

Comment:
As we mention in command documentation, let us clarify the scaling and non scaling cases where expansion is nil.